Packet++/src/Packet.cpp

```diff
@@ -771,6 +772,10 @@ Layer* Packet::createFirstLayer(LinkLayerType linkType)
 			? static_cast<Layer*>(new IPv6Layer((uint8_t*)rawData, rawDataLen, nullptr, this))
 			: static_cast<Layer*>(new PayloadLayer((uint8_t*)rawData, rawDataLen, nullptr, this));
 	}
+	else if (linkType == LINKTYPE_NFLOG)
+	{
+		return new NflogLayer((uint8_t*)rawData, rawDataLen, this);
```

Comment:
I think we should add a static method:
```cpp
/**
 * A static method that validates the input data
 * @param[in] data The pointer to the beginning of a byte stream of an NFLOG packet
 * @param[in] dataLen The length of the byte stream
 * @return True if the data is valid and can represent an NFLOG packet
 */
static bool isDataValid(const uint8_t* data, size_t dataLen)
{
	return data && dataLen >= sizeof(nflog_header);
}
```

Then we can do here as we do in other places:
```cpp
else if (linkType == LINKTYPE_NFLOG)
{
	return NflogLayer::isDataValid(rawData, rawDataLen)
		? static_cast<Layer*>(new NflogLayer((uint8_t*)rawData, rawDataLen, this))
		: static_cast<Layer*>(new PayloadLayer((uint8_t*)rawData, rawDataLen, nullptr, this));
```